src/util/backend.js

```diff
@@ -0,0 +1,313 @@
+const deepEqual = require('deep-equal')
+const BigNumber = require('bignumber.js')
+const KEY_REGEX = /^[A-Za-z0-9\-_]*$/
+
+class ObjTransferLog {
+  constructor (store, opts) {
+    this.maximum = new BigNumber(opts.maximum)
+    this.minimum = new BigNumber(opts.minimum)
+    this.cache = {}
+    this.key = opts.key || ''
+    if (!this.key.match(KEY_REGEX)) {
+      throw new Error('invalid key: ' + this.key)
+    }
+
+    // optional, for stateful plugin only
+    this.store = store
+    this.writeQueue = Promise.resolve()
+
+    // TODO: disable balance? (not needed for client plugin)
+    this.balance_if = new BigNumber(0)
```

Comment:
Please replace `balance_if` with whatever `if` stands for -- makes it much easier to read and understand later! Also shouldn't all variables use camel instead of snake case?